tuning/config/acceleration_configs/utils.py

```diff
@@ -26,12 +27,15 @@ def ensure_nested_dataclasses_initialized(dataclass: DataClass):
     this is to be called at the top-level class to init all the
     nested dataclasses.
     """
-    type_hints: Dict[str, type] = get_type_hints(dataclass)
+    type_hints = get_type_hints(dataclass)
     for f in fields(dataclass):
         nested_type = type_hints[f.name]
         values = getattr(dataclass, f.name)
         if values is not None and not is_dataclass(values):
-            values = nested_type(*values)
+            if isinstance(values, Iterable) and not isinstance(values, (str, bytes)):
```

Comment:
@willmj can explain this fix? in this PR we are not doing anything much different then before, so why is this needed?

Reply:
Since there's only one value in FastMoeConfig that's an int, if the previous implementation tried to unpack it, it caused this error:
```
Traceback (most recent call last):
  File "/home/tuning/.local/lib/python3.11/site-packages/tuning/sft_trainer.py", line 601, in main
    ) = parse_arguments(parser, job_config)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tuning/.local/lib/python3.11/site-packages/tuning/sft_trainer.py", line 535, in parse_arguments
    ) = parser.parse_dict(json_config, allow_extra_keys=True)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/tuning/.local/lib/python3.11/site-packages/transformers/hf_argparser.py", line 374, in parse_dict
    obj = dtype(**inputs)
          ^^^^^^^^^^^^^^^
  File "<string>", line 4, in __init__
  File "/home/tuning/.local/lib/python3.11/site-packages/tuning/config/acceleration_configs/fast_moe.py", line 36, in __post_init__
    ensure_nested_dataclasses_initialized(self)
  File "/home/tuning/.local/lib/python3.11/site-packages/tuning/config/acceleration_configs/utils.py", line 34, in ensure_nested_dataclasses_initialized
    values = nested_type(*values)
             ^^^^^^^^^^^^^^^^^^^^
TypeError: tuning.config.acceleration_configs.utils.parsable_dataclass.<locals>.ParsableDataclass() argument after * must be an iterable, not int
```
So this seemed like the best way to avoid that. If you have another solution though let me know.

Reply:
@willmj thanks for the explaination. this fix seems reasonable, but i would suggest putting comments to explain this in the code